main/src/main/scala/sbt/Defaults.scala

```diff
@@ -661,7 +663,7 @@ object Defaults extends BuildCommon {
     classDirectory := target.value / (prefix(configuration.value.name) + "classes"),
     backendOutput := {
       val converter = fileConverter.value
-      val dir = classDirectory.value
+      val dir = target.value / (prefix(configuration.value.name) + "backend-output")
```

Comment:
`classDirectory` was:
  - the folder where we compile classes to
  - the folder where we zip classes from
  - the folder where we copy resources to
  - the folder where we unzip the compiled jar to
  - the folder where we zip classes and resources from
  
And this was a source of bugs: old resource files being unzipped on top of new resource files. In practice I think it is simpler and safer if `backendOutput` (the output directory of the compiler) and `classDirectory` (where we extract classes and copy resources) are different folders.